libs/isograph-react/src/core/check.ts

```diff
@@ -151,8 +146,7 @@ function checkFromRecord(
               normalizationAstNode.selections,
               variables,
               linkedRecord,
-              // TODO this seems likely to be wrong
-              backupId + '.' + parentRecordKey,
+              link,
```

Comment:
here, and in other place `linkedRecord` is `environment.store[link.__link]`, so `link` makes sense here